seq_tools/validation/c670_rg_is_paired_in_bam.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+
+"""
+    Copyright (c) 2020, Ontario Institute for Cancer Research (OICR).
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published
+    by the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+    You should have received a copy of the GNU Affero General Public License
+    along with this program. If not, see <https://www.gnu.org/licenses/>.
+    Authors:
+        Junjun Zhang <junjun.zhang@oicr.on.ca>
+        Linda Xiang <linda.xiang@oicr.on.ca>
+        Edmund Su <edmund.su@oicr.on.ca>
+"""
+
+
+import os
+from base_checker import BaseChecker
+import subprocess
+import re
+
+
+class Checker(BaseChecker):
+    def __init__(self, ctx, metadata, skip=False):
+        super().__init__(ctx, metadata, __name__, skip=skip)
+
+    @BaseChecker._catch_exception
+    def check(self):
+        # status already set at initiation
+        if self.status:
+            return
+
+        if not self.metadata.get('read_groups'):
+            message = "Missing 'read_groups' section in the metadata JSON"
+            self.logger.info(f'[{self.checker}] {message}')
+            self.message = message
+            self.status = 'INVALID'
+            return
+    
+            
+        offending_rgs ={}
+        for rg in self.metadata.get('read_groups'):
+        
+            if 'is_paired_end' not in rg:
+                message = "'is_pair_end' not found in readgroup : %s" % rg['submitter_read_group_id']
+                self.logger.info(f'[{self.checker}] {message}')
+                self.message = message
+                self.status = 'INVALID'
+                return
+            elif rg['is_paired_end']==None:
+                message = "'is_pair_end' value in readgroup is null. Must be boolean: %s" % rg['submitter_read_group_id']
```

Comment:
```
" Field 'is_paired_end' for read_group: %s in metadata JSON is null. Must be boolean." % rg['submitter_read_group_id']
```

Reply:
same question, what will be reported for this use case?

Reply:
Added test case to demonstrate report
https://github.com/icgc-argo/seq-tools/blob/55a34779f106549fd3809cc495c4755169174db5/tests/expected_reports/anon_chr1_rnaseq/anon_chr1_meta_ispaired_null.validation_report.INVALID.jsonl#L22